src/main/java/eu/dissco/nusearch/component/MessageCompressionComponent.java

```diff
@@ -0,0 +1,78 @@
+package eu.dissco.nusearch.component;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageProperties;
+import org.springframework.amqp.support.converter.MessageConversionException;
+import org.springframework.amqp.support.converter.MessageConverter;
+import org.springframework.amqp.support.converter.SimpleMessageConverter;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class MessageCompressionComponent implements MessageConverter {
+
+  final MessageConverter simpleConverter = new SimpleMessageConverter();
+
+  private static byte[] deflateMessage(byte[] message) throws IOException {
+    final int BUFFER_SIZE = 8192; // 8KB
+    try (var rstBao = new ByteArrayOutputStream(BUFFER_SIZE)) {
```

Comment:
put both resources on one line?

Reply:
See other comment on translator, this is necessary.